2.0.0-alpha.8

matteobruni opened the pull request.

matteobruni committed:
build: removed unused imports

matteobruni committed:
feat: preparing simpler options for confetti preset

matteobruni committed:
feat: prepared confetti preset

matteobruni committed:
docs: improving documentation

dependabot[bot] committed:
build(deps-dev): bump rollup from 2.48.0 to 2.49.0

Bumps [rollup](https://github.com/rollup/rollup) from 2.48.0 to 2.49.0.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v2.48.0...v2.49.0)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump ts-loader from 9.2.1 to 9.2.2

Bumps [ts-loader](https://github.com/TypeStrong/ts-loader) from 9.2.1 to 9.2.2.
- [Release notes](https://github.com/TypeStrong/ts-loader/releases)
- [Changelog](https://github.com/TypeStrong/ts-loader/blob/main/CHANGELOG.md)
- [Commits](https://github.com/TypeStrong/ts-loader/compare/v9.2.1...v9.2.2)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1551 from matteobruni/dependabot/npm_and_yarn/dev/rollup-2.49.0

build(deps-dev): bump rollup from 2.48.0 to 2.49.0

matteobruni committed:
Merge pull request #1552 from matteobruni/dependabot/npm_and_yarn/dev/ts-loader-9.2.2

build(deps-dev): bump ts-loader from 9.2.1 to 9.2.2

matteobruni committed:
docs: improving documentation

dependabot[bot] committed:
build(deps): bump @angular/forms from 12.0.0 to 12.0.1

Bumps [@angular/forms](https://github.com/angular/angular/tree/HEAD/packages/forms) from 12.0.0 to 12.0.1.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.1/packages/forms)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump ts-node from 9.1.1 to 10.0.0

Bumps [ts-node](https://github.com/TypeStrong/ts-node) from 9.1.1 to 10.0.0.
- [Release notes](https://github.com/TypeStrong/ts-node/releases)
- [Commits](https://github.com/TypeStrong/ts-node/compare/v9.1.1...v10.0.0)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1554 from matteobruni/dependabot/npm_and_yarn/dev/ts-node-10.0.0

build(deps-dev): bump ts-node from 9.1.1 to 10.0.0

matteobruni committed:
docs: improving documentation

matteobruni committed:
build: deps updated

dependabot[bot] committed:
build(deps-dev): bump svelte-check from 1.5.4 to 1.6.0

Bumps [svelte-check](https://github.com/sveltejs/language-tools) from 1.5.4 to 1.6.0.
- [Release notes](https://github.com/sveltejs/language-tools/releases)
- [Commits](https://github.com/sveltejs/language-tools/compare/svelte-check-1.5.4...svelte-check-1.6.0)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1555 from matteobruni/dependabot/npm_and_yarn/dev/svelte-check-1.6.0

build(deps-dev): bump svelte-check from 1.5.4 to 1.6.0

matteobruni committed:
feat: adding SolidJS component

dependabot[bot] committed:
build(deps): bump browserslist in /components/preact

Bumps [browserslist](https://github.com/browserslist/browserslist) from 4.16.3 to 4.16.6.
- [Release notes](https://github.com/browserslist/browserslist/releases)
- [Changelog](https://github.com/browserslist/browserslist/blob/main/CHANGELOG.md)
- [Commits](https://github.com/browserslist/browserslist/compare/4.16.3...4.16.6)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump browserslist in /components/inferno

Bumps [browserslist](https://github.com/browserslist/browserslist) from 4.16.3 to 4.16.6.
- [Release notes](https://github.com/browserslist/browserslist/releases)
- [Changelog](https://github.com/browserslist/browserslist/blob/main/CHANGELOG.md)
- [Commits](https://github.com/browserslist/browserslist/compare/4.16.3...4.16.6)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump browserslist in /components/react

Bumps [browserslist](https://github.com/browserslist/browserslist) from 4.16.3 to 4.16.6.
- [Release notes](https://github.com/browserslist/browserslist/releases)
- [Changelog](https://github.com/browserslist/browserslist/blob/main/CHANGELOG.md)
- [Commits](https://github.com/browserslist/browserslist/compare/4.16.3...4.16.6)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump browserslist from 4.16.3 to 4.16.6 in /components/vue3

Bumps [browserslist](https://github.com/browserslist/browserslist) from 4.16.3 to 4.16.6.
- [Release notes](https://github.com/browserslist/browserslist/releases)
- [Changelog](https://github.com/browserslist/browserslist/blob/main/CHANGELOG.md)
- [Commits](https://github.com/browserslist/browserslist/compare/4.16.3...4.16.6)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump browserslist from 4.16.3 to 4.16.6 in /components/vue

Bumps [browserslist](https://github.com/browserslist/browserslist) from 4.16.3 to 4.16.6.
- [Release notes](https://github.com/browserslist/browserslist/releases)
- [Changelog](https://github.com/browserslist/browserslist/blob/main/CHANGELOG.md)
- [Commits](https://github.com/browserslist/browserslist/compare/4.16.3...4.16.6)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump browserslist in /components/angular

Bumps [browserslist](https://github.com/browserslist/browserslist) from 4.16.3 to 4.16.6.
- [Release notes](https://github.com/browserslist/browserslist/releases)
- [Changelog](https://github.com/browserslist/browserslist/blob/main/CHANGELOG.md)
- [Commits](https://github.com/browserslist/browserslist/compare/4.16.3...4.16.6)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1557 from matteobruni/dependabot/npm_and_yarn/components/preact/browserslist-4.16.6

build(deps): bump browserslist from 4.16.3 to 4.16.6 in /components/preact

matteobruni committed:
Merge branch 'main' into dependabot/npm_and_yarn/components/inferno/browserslist-4.16.6

matteobruni committed:
Merge branch 'main' into dependabot/npm_and_yarn/components/react/browserslist-4.16.6

matteobruni committed:
Merge branch 'main' into dependabot/npm_and_yarn/components/vue3/browserslist-4.16.6

matteobruni committed:
Merge branch 'main' into dependabot/npm_and_yarn/components/vue/browserslist-4.16.6

matteobruni committed:
Merge branch 'main' into dependabot/npm_and_yarn/components/angular/browserslist-4.16.6

dependabot[bot] committed:
build(deps-dev): bump @angular/cli from 12.0.0 to 12.0.2

Bumps [@angular/cli](https://github.com/angular/angular-cli) from 12.0.0 to 12.0.2.
- [Release notes](https://github.com/angular/angular-cli/releases)
- [Commits](https://github.com/angular/angular-cli/compare/v12.0.0...v12.0.2)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1558 from matteobruni/dependabot/npm_and_yarn/components/inferno/browserslist-4.16.6

build(deps): bump browserslist from 4.16.3 to 4.16.6 in /components/inferno

matteobruni committed:
Merge branch 'main' into dependabot/npm_and_yarn/components/react/browserslist-4.16.6

matteobruni committed:
Merge pull request #1559 from matteobruni/dependabot/npm_and_yarn/components/react/browserslist-4.16.6

build(deps): bump browserslist from 4.16.3 to 4.16.6 in /components/react

matteobruni committed:
Merge branch 'main' into dependabot/npm_and_yarn/components/vue3/browserslist-4.16.6

matteobruni committed:
Merge pull request #1560 from matteobruni/dependabot/npm_and_yarn/components/vue3/browserslist-4.16.6

build(deps): bump browserslist from 4.16.3 to 4.16.6 in /components/vue3

matteobruni committed:
Merge branch 'main' into dependabot/npm_and_yarn/components/vue/browserslist-4.16.6

matteobruni committed:
Merge pull request #1561 from matteobruni/dependabot/npm_and_yarn/components/vue/browserslist-4.16.6

build(deps): bump browserslist from 4.16.3 to 4.16.6 in /components/vue

matteobruni committed:
Merge branch 'main' into dependabot/npm_and_yarn/components/angular/browserslist-4.16.6

dependabot[bot] committed:
build(deps): bump dns-packet from 1.3.1 to 1.3.4 in /components/vue

Bumps [dns-packet](https://github.com/mafintosh/dns-packet) from 1.3.1 to 1.3.4.
- [Release notes](https://github.com/mafintosh/dns-packet/releases)
- [Changelog](https://github.com/mafintosh/dns-packet/blob/master/CHANGELOG.md)
- [Commits](https://github.com/mafintosh/dns-packet/compare/v1.3.1...v1.3.4)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1562 from matteobruni/dependabot/npm_and_yarn/components/angular/browserslist-4.16.6

build(deps): bump browserslist from 4.16.3 to 4.16.6 in /components/angular

matteobruni committed:
Merge pull request #1563 from matteobruni/dependabot/npm_and_yarn/dev/angular/cli-12.0.2

build(deps-dev): bump @angular/cli from 12.0.0 to 12.0.2

matteobruni committed:
Merge branch 'main' into dependabot/npm_and_yarn/components/vue/dns-packet-1.3.4

dependabot[bot] committed:
build(deps): bump dns-packet from 1.3.1 to 1.3.4 in /components/inferno

Bumps [dns-packet](https://github.com/mafintosh/dns-packet) from 1.3.1 to 1.3.4.
- [Release notes](https://github.com/mafintosh/dns-packet/releases)
- [Changelog](https://github.com/mafintosh/dns-packet/blob/master/CHANGELOG.md)
- [Commits](https://github.com/mafintosh/dns-packet/compare/v1.3.1...v1.3.4)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1564 from matteobruni/dependabot/npm_and_yarn/components/vue/dns-packet-1.3.4

build(deps): bump dns-packet from 1.3.1 to 1.3.4 in /components/vue

matteobruni committed:
Merge branch 'main' into dependabot/npm_and_yarn/components/inferno/dns-packet-1.3.4

dependabot[bot] committed:
build(deps): bump dns-packet from 1.3.1 to 1.3.4 in /components/vue3

Bumps [dns-packet](https://github.com/mafintosh/dns-packet) from 1.3.1 to 1.3.4.
- [Release notes](https://github.com/mafintosh/dns-packet/releases)
- [Changelog](https://github.com/mafintosh/dns-packet/blob/master/CHANGELOG.md)
- [Commits](https://github.com/mafintosh/dns-packet/compare/v1.3.1...v1.3.4)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1567 from matteobruni/dependabot/npm_and_yarn/components/inferno/dns-packet-1.3.4

build(deps): bump dns-packet from 1.3.1 to 1.3.4 in /components/inferno

matteobruni committed:
Merge branch 'main' into dependabot/npm_and_yarn/components/vue3/dns-packet-1.3.4

dependabot[bot] committed:
build(deps): bump dns-packet from 1.3.1 to 1.3.4 in /components/angular

Bumps [dns-packet](https://github.com/mafintosh/dns-packet) from 1.3.1 to 1.3.4.
- [Release notes](https://github.com/mafintosh/dns-packet/releases)
- [Changelog](https://github.com/mafintosh/dns-packet/blob/master/CHANGELOG.md)
- [Commits](https://github.com/mafintosh/dns-packet/compare/v1.3.1...v1.3.4)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1565 from matteobruni/dependabot/npm_and_yarn/components/vue3/dns-packet-1.3.4

build(deps): bump dns-packet from 1.3.1 to 1.3.4 in /components/vue3

matteobruni committed:
Merge branch 'main' into dependabot/npm_and_yarn/components/angular/dns-packet-1.3.4

matteobruni committed:
Merge pull request #1566 from matteobruni/dependabot/npm_and_yarn/components/angular/dns-packet-1.3.4

build(deps): bump dns-packet from 1.3.1 to 1.3.4 in /components/angular

dependabot[bot] committed:
build(deps): bump @angular/common from 12.0.0 to 12.0.2

Bumps [@angular/common](https://github.com/angular/angular/tree/HEAD/packages/common) from 12.0.0 to 12.0.2.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.2/packages/common)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/animations from 12.0.0 to 12.0.2

Bumps [@angular/animations](https://github.com/angular/angular/tree/HEAD/packages/animations) from 12.0.0 to 12.0.2.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.2/packages/animations)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1568 from matteobruni/dependabot/npm_and_yarn/dev/angular/common-12.0.2

build(deps): bump @angular/common from 12.0.0 to 12.0.2

matteobruni committed:
Merge pull request #1569 from matteobruni/dependabot/npm_and_yarn/dev/angular/animations-12.0.2

build(deps): bump @angular/animations from 12.0.0 to 12.0.2

matteobruni committed:
chore(solid): improving solid component, now project builds

matteobruni committed:
chore(solid): improving solid component, now project builds

matteobruni committed:
chore(solid): improving solid component

matteobruni committed:
chore(solid): improving solid component

matteobruni committed:
chore(solid): improving solid component

matteobruni committed:
chore(solid): improving solid component

matteobruni committed:
chore(solid): improving solid component

matteobruni committed:
chore(solid): improving solid component

matteobruni committed:
chore(solid): improving solid component

matteobruni committed:
feat: added web component

matteobruni committed:
Merge pull request #1553 from matteobruni/dependabot/npm_and_yarn/dev/angular/forms-12.0.1

build(deps): bump @angular/forms from 12.0.0 to 12.0.1

matteobruni committed:
chore(component): improved web component

matteobruni committed:
build: preparing web components demo

dependabot[bot] committed:
build(deps): bump ws from 7.4.4 to 7.4.6 in /components/react

Bumps [ws](https://github.com/websockets/ws) from 7.4.4 to 7.4.6.
- [Release notes](https://github.com/websockets/ws/releases)
- [Commits](https://github.com/websockets/ws/compare/7.4.4...7.4.6)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump ws from 7.4.4 to 7.4.6 in /components/preact

Bumps [ws](https://github.com/websockets/ws) from 7.4.4 to 7.4.6.
- [Release notes](https://github.com/websockets/ws/releases)
- [Commits](https://github.com/websockets/ws/compare/7.4.4...7.4.6)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump rollup from 2.49.0 to 2.50.5

Bumps [rollup](https://github.com/rollup/rollup) from 2.49.0 to 2.50.5.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v2.49.0...v2.50.5)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump bootstrap from 4.6.0 to 5.0.1

Bumps [bootstrap](https://github.com/twbs/bootstrap) from 4.6.0 to 5.0.1.
- [Release notes](https://github.com/twbs/bootstrap/releases)
- [Commits](https://github.com/twbs/bootstrap/compare/v4.6.0...v5.0.1)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1571 from matteobruni/dependabot/npm_and_yarn/components/react/ws-7.4.6

build(deps): bump ws from 7.4.4 to 7.4.6 in /components/react

matteobruni committed:
Merge branch 'main' into dependabot/npm_and_yarn/components/preact/ws-7.4.6

matteobruni committed:
Merge pull request #1573 from matteobruni/dependabot/npm_and_yarn/dev/rollup-2.50.5

build(deps-dev): bump rollup from 2.49.0 to 2.50.5

matteobruni committed:
Merge pull request #1574 from matteobruni/dependabot/npm_and_yarn/dev/bootstrap-5.0.1

build(deps-dev): bump bootstrap from 4.6.0 to 5.0.1

matteobruni committed:
build: improving web component

matteobruni committed:
Merge pull request #1572 from matteobruni/dependabot/npm_and_yarn/components/preact/ws-7.4.6

build(deps): bump ws from 7.4.4 to 7.4.6 in /components/preact

matteobruni committed:
build: fixed web component

matteobruni committed:
Merge pull request #1570 from matteobruni/web-component

feat: added web component

matteobruni committed:
build: improving web component

matteobruni committed:
build: improving canvas resize

matteobruni committed:
build: preparing web-component for publishing

matteobruni committed:
build: preparing web-component for publishing

dependabot[bot] committed:
build(deps): bump rxjs from 7.0.1 to 7.1.0

Bumps [rxjs](https://github.com/reactivex/rxjs) from 7.0.1 to 7.1.0.
- [Release notes](https://github.com/reactivex/rxjs/releases)
- [Changelog](https://github.com/ReactiveX/rxjs/blob/master/CHANGELOG.md)
- [Commits](https://github.com/reactivex/rxjs/compare/7.0.1...7.1.0)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @commitlint/cli from 12.1.1 to 12.1.4

Bumps [@commitlint/cli](https://github.com/conventional-changelog/commitlint) from 12.1.1 to 12.1.4.
- [Release notes](https://github.com/conventional-changelog/commitlint/releases)
- [Changelog](https://github.com/conventional-changelog/commitlint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/conventional-changelog/commitlint/compare/v12.1.1...v12.1.4)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/react from 17.0.6 to 17.0.8

Bumps [@types/react](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/react) from 17.0.6 to 17.0.8.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/react)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1576 from matteobruni/dependabot/npm_and_yarn/dev/rxjs-7.1.0

build(deps): bump rxjs from 7.0.1 to 7.1.0

matteobruni committed:
Merge pull request #1577 from matteobruni/dependabot/npm_and_yarn/dev/commitlint/cli-12.1.4

build(deps-dev): bump @commitlint/cli from 12.1.1 to 12.1.4

matteobruni committed:
Merge pull request #1578 from matteobruni/dependabot/npm_and_yarn/dev/types/react-17.0.8

build(deps-dev): bump @types/react from 17.0.6 to 17.0.8

dependabot[bot] committed:
build(deps): bump riot from 5.3.3 to 5.4.3

Bumps [riot](https://github.com/riot/riot) from 5.3.3 to 5.4.3.
- [Release notes](https://github.com/riot/riot/releases)
- [Commits](https://github.com/riot/riot/compare/v5.3.3...v5.4.3)

---
updated-dependencies:
- dependency-name: riot
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/react from 17.0.8 to 17.0.9

Bumps [@types/react](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/react) from 17.0.8 to 17.0.9.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/react)

---
updated-dependencies:
- dependency-name: "@types/react"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump eslint-plugin-vue from 7.9.0 to 7.10.0

Bumps [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue) from 7.9.0 to 7.10.0.
- [Release notes](https://github.com/vuejs/eslint-plugin-vue/releases)
- [Commits](https://github.com/vuejs/eslint-plugin-vue/compare/v7.9.0...v7.10.0)

---
updated-dependencies:
- dependency-name: eslint-plugin-vue
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1579 from matteobruni/dependabot/npm_and_yarn/dev/riot-5.4.3

build(deps): bump riot from 5.3.3 to 5.4.3

matteobruni committed:
Merge pull request #1580 from matteobruni/dependabot/npm_and_yarn/dev/types/react-17.0.9

build(deps-dev): bump @types/react from 17.0.8 to 17.0.9

matteobruni committed:
Merge pull request #1581 from matteobruni/dependabot/npm_and_yarn/dev/eslint-plugin-vue-7.10.0

build(deps-dev): bump eslint-plugin-vue from 7.9.0 to 7.10.0

dependabot[bot] committed:
build(deps): bump ws from 6.2.1 to 6.2.2 in /components/vue3

Bumps [ws](https://github.com/websockets/ws) from 6.2.1 to 6.2.2.
- [Release notes](https://github.com/websockets/ws/releases)
- [Commits](https://github.com/websockets/ws/commits)

---
updated-dependencies:
- dependency-name: ws
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1582 from matteobruni/dependabot/npm_and_yarn/components/vue3/ws-6.2.2

build(deps): bump ws from 6.2.1 to 6.2.2 in /components/vue3

dependabot[bot] committed:
build(deps): bump ws from 6.2.1 to 6.2.2 in /components/inferno

Bumps [ws](https://github.com/websockets/ws) from 6.2.1 to 6.2.2.
- [Release notes](https://github.com/websockets/ws/releases)
- [Commits](https://github.com/websockets/ws/commits)

---
updated-dependencies:
- dependency-name: ws
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump ws from 6.2.1 to 6.2.2 in /components/vue

Bumps [ws](https://github.com/websockets/ws) from 6.2.1 to 6.2.2.
- [Release notes](https://github.com/websockets/ws/releases)
- [Commits](https://github.com/websockets/ws/commits)

---
updated-dependencies:
- dependency-name: ws
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1583 from matteobruni/dependabot/npm_and_yarn/components/inferno/ws-6.2.2

build(deps): bump ws from 6.2.1 to 6.2.2 in /components/inferno

matteobruni committed:
Merge branch 'main' into dependabot/npm_and_yarn/components/vue/ws-6.2.2

dependabot[bot] committed:
build(deps): bump ws from 6.2.1 to 6.2.2 in /components/angular

Bumps [ws](https://github.com/websockets/ws) from 6.2.1 to 6.2.2.
- [Release notes](https://github.com/websockets/ws/releases)
- [Commits](https://github.com/websockets/ws/commits)

---
updated-dependencies:
- dependency-name: ws
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1584 from matteobruni/dependabot/npm_and_yarn/components/vue/ws-6.2.2

build(deps): bump ws from 6.2.1 to 6.2.2 in /components/vue

matteobruni committed:
Merge branch 'main' into dependabot/npm_and_yarn/components/angular/ws-6.2.2

matteobruni committed:
Merge pull request #1585 from matteobruni/dependabot/npm_and_yarn/components/angular/ws-6.2.2

build(deps): bump ws from 6.2.1 to 6.2.2 in /components/angular

matteobruni committed:
Merge branch 'v1' into solid-js

dependabot[bot] committed:
build(deps-dev): bump @types/node from 15.3.0 to 15.9.0

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 15.3.0 to 15.9.0.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/common from 12.0.2 to 12.0.3

Bumps [@angular/common](https://github.com/angular/angular/tree/HEAD/packages/common) from 12.0.2 to 12.0.3.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.3/packages/common)

---
updated-dependencies:
- dependency-name: "@angular/common"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/platform-browser-dynamic

Bumps [@angular/platform-browser-dynamic](https://github.com/angular/angular/tree/HEAD/packages/platform-browser-dynamic) from 12.0.0 to 12.0.3.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.3/packages/platform-browser-dynamic)

---
updated-dependencies:
- dependency-name: "@angular/platform-browser-dynamic"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1588 from matteobruni/dependabot/npm_and_yarn/dev/types/node-15.9.0

build(deps-dev): bump @types/node from 15.3.0 to 15.9.0

matteobruni committed:
Merge pull request #1589 from matteobruni/dependabot/npm_and_yarn/dev/angular/common-12.0.3

build(deps): bump @angular/common from 12.0.2 to 12.0.3

matteobruni committed:
Merge pull request #1590 from matteobruni/dependabot/npm_and_yarn/dev/angular/platform-browser-dynamic-12.0.3

build(deps): bump @angular/platform-browser-dynamic from 12.0.0 to 12.0.3

dependabot[bot] committed:
build(deps): bump @angular/router from 12.0.0 to 12.0.3

Bumps [@angular/router](https://github.com/angular/angular/tree/HEAD/packages/router) from 12.0.0 to 12.0.3.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.3/packages/router)

---
updated-dependencies:
- dependency-name: "@angular/router"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump canvas from 2.7.0 to 2.8.0

Bumps [canvas](https://github.com/Automattic/node-canvas) from 2.7.0 to 2.8.0.
- [Release notes](https://github.com/Automattic/node-canvas/releases)
- [Changelog](https://github.com/Automattic/node-canvas/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Automattic/node-canvas/compare/v2.7.0...v2.8.0)

---
updated-dependencies:
- dependency-name: canvas
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/core from 12.0.0 to 12.0.3

Bumps [@angular/core](https://github.com/angular/angular/tree/HEAD/packages/core) from 12.0.0 to 12.0.3.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.3/packages/core)

---
updated-dependencies:
- dependency-name: "@angular/core"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1591 from matteobruni/dependabot/npm_and_yarn/dev/angular/router-12.0.3

build(deps): bump @angular/router from 12.0.0 to 12.0.3

matteobruni committed:
Merge pull request #1592 from matteobruni/dependabot/npm_and_yarn/dev/canvas-2.8.0

build(deps-dev): bump canvas from 2.7.0 to 2.8.0

matteobruni committed:
Merge pull request #1593 from matteobruni/dependabot/npm_and_yarn/dev/angular/core-12.0.3

build(deps): bump @angular/core from 12.0.0 to 12.0.3

matteobruni committed:
chore(components): fixed solid component

matteobruni committed:
Merge pull request #1556 from matteobruni/solid-js

feat: adding SolidJS component

matteobruni committed:
build: updated angular component deps

matteobruni committed:
build: updated inferno and jquery component deps

matteobruni committed:
build: updated preact component deps

matteobruni committed:
build: updated solid component deps

matteobruni committed:
build: updated react component deps

matteobruni committed:
build: updated riot component deps

matteobruni committed:
build: updated svelte component deps

matteobruni committed:
build: updated vue component deps

matteobruni committed:
build: updated web component deps

matteobruni committed:
build: updated core packages deps, broken typedoc with ts4.3

matteobruni committed:
build: updated presets and shapes deps

matteobruni committed:
build: updated templates

matteobruni committed:
build: updated angular demo deps

matteobruni committed:
build: updaed editor demo deps

matteobruni committed:
build: updaed inferno demo deps

matteobruni committed:
build: updaed jquery demo deps

matteobruni committed:
build: updaed preact demo deps

matteobruni committed:
build: updaed react demo deps

matteobruni committed:
build: updaed riot demo deps

matteobruni committed:
build: updaed solid demo deps

matteobruni committed:
build: updaed svelte demo deps

matteobruni committed:
build: updaed vanilla and vue demo deps

matteobruni committed:
build: updated web components demo deps

matteobruni committed:
Merge branch 'main' into v1

matteobruni committed:
build: updates after merge

matteobruni committed:
fix: fixed issue #1594

matteobruni committed:
Merge pull request #1596 from matteobruni/issue-1594

Fix Issue 1594

dependabot[bot] committed:
build(deps-dev): bump ng-packagr from 12.0.0 to 12.0.3

Bumps [ng-packagr](https://github.com/ng-packagr/ng-packagr) from 12.0.0 to 12.0.3.
- [Release notes](https://github.com/ng-packagr/ng-packagr/releases)
- [Changelog](https://github.com/ng-packagr/ng-packagr/blob/master/CHANGELOG.md)
- [Commits](https://github.com/ng-packagr/ng-packagr/compare/v12.0.0...v12.0.3)

---
updated-dependencies:
- dependency-name: ng-packagr
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @tsconfig/svelte from 1.0.10 to 2.0.1

Bumps [@tsconfig/svelte](https://github.com/tsconfig/bases/tree/HEAD/bases) from 1.0.10 to 2.0.1.
- [Release notes](https://github.com/tsconfig/bases/releases)
- [Commits](https://github.com/tsconfig/bases/commits/HEAD/bases)

---
updated-dependencies:
- dependency-name: "@tsconfig/svelte"
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump source-map-loader from 2.0.1 to 3.0.0

Bumps [source-map-loader](https://github.com/webpack-contrib/source-map-loader) from 2.0.1 to 3.0.0.
- [Release notes](https://github.com/webpack-contrib/source-map-loader/releases)
- [Changelog](https://github.com/webpack-contrib/source-map-loader/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack-contrib/source-map-loader/compare/v2.0.1...v3.0.0)

---
updated-dependencies:
- dependency-name: source-map-loader
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
fix: debounce window resize, check if #1587 is fixed
feat: added firefly preset

matteobruni committed:
fix: fixed #1587

matteobruni committed:
Merge pull request #1597 from matteobruni/dependabot/npm_and_yarn/dev/ng-packagr-12.0.3

build(deps-dev): bump ng-packagr from 12.0.0 to 12.0.3

matteobruni committed:
Merge pull request #1598 from matteobruni/dependabot/npm_and_yarn/dev/tsconfig/svelte-2.0.1

build(deps-dev): bump @tsconfig/svelte from 1.0.10 to 2.0.1

matteobruni committed:
Merge pull request #1599 from matteobruni/dependabot/npm_and_yarn/dev/source-map-loader-3.0.0

build(deps-dev): bump source-map-loader from 2.0.1 to 3.0.0

matteobruni committed:
build: preparing docs for release

matteobruni committed:
build: preparing docs for release

matteobruni committed:
build: preparing docs for release

matteobruni committed:
build: preparing docs for release

matteobruni committed:
build: preparing docs for release
chore(solid): the local component was used, the external one is still broken

matteobruni committed:
build: solid component is now private until a fix is found, so the 1.28.0 can be released

matteobruni committed:
build: web components demo has less files

matteobruni committed:
chore(firefly): improving firefly preset

matteobruni committed:
build: improving docs for release

matteobruni committed:
build: try fix solid

matteobruni committed:
build: fixed solid component

dependabot[bot] committed:
build(deps-dev): bump jsdom from 16.5.3 to 16.6.0

Bumps [jsdom](https://github.com/jsdom/jsdom) from 16.5.3 to 16.6.0.
- [Release notes](https://github.com/jsdom/jsdom/releases)
- [Changelog](https://github.com/jsdom/jsdom/blob/master/Changelog.md)
- [Commits](https://github.com/jsdom/jsdom/compare/16.5.3...16.6.0)

---
updated-dependencies:
- dependency-name: jsdom
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump webpack-cli from 4.7.0 to 4.7.2

Bumps [webpack-cli](https://github.com/webpack/webpack-cli) from 4.7.0 to 4.7.2.
- [Release notes](https://github.com/webpack/webpack-cli/releases)
- [Changelog](https://github.com/webpack/webpack-cli/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack/webpack-cli/compare/webpack-cli@4.7.0...webpack-cli@4.7.2)

---
updated-dependencies:
- dependency-name: webpack-cli
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @angular/compiler-cli from 12.0.0 to 12.0.3

Bumps [@angular/compiler-cli](https://github.com/angular/angular/tree/HEAD/packages/compiler-cli) from 12.0.0 to 12.0.3.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.3/packages/compiler-cli)

---
updated-dependencies:
- dependency-name: "@angular/compiler-cli"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1602 from matteobruni/dependabot/npm_and_yarn/dev/angular/compiler-cli-12.0.3

build(deps-dev): bump @angular/compiler-cli from 12.0.0 to 12.0.3

matteobruni committed:
Merge pull request #1600 from matteobruni/dependabot/npm_and_yarn/dev/jsdom-16.6.0

build(deps-dev): bump jsdom from 16.5.3 to 16.6.0

matteobruni committed:
Merge pull request #1601 from matteobruni/dependabot/npm_and_yarn/dev/webpack-cli-4.7.2

build(deps-dev): bump webpack-cli from 4.7.0 to 4.7.2

matteobruni committed:
build: updated some preset demo images

matteobruni committed:
fix: fixed confetti preset
build: fixed confetti preset docs

matteobruni committed:
build: added fire preset docs

dependabot[bot] committed:
build(deps-dev): bump @riotjs/cli from 5.2.1 to 5.3.0

Bumps [@riotjs/cli](https://github.com/riot/cli) from 5.2.1 to 5.3.0.
- [Release notes](https://github.com/riot/cli/releases)
- [Changelog](https://github.com/riot/cli/blob/main/CHANGELOG.md)
- [Commits](https://github.com/riot/cli/compare/v5.2.1...v5.3.0)

---
updated-dependencies:
- dependency-name: "@riotjs/cli"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/platform-browser from 12.0.0 to 12.0.3

Bumps [@angular/platform-browser](https://github.com/angular/angular/tree/HEAD/packages/platform-browser) from 12.0.0 to 12.0.3.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.3/packages/platform-browser)

---
updated-dependencies:
- dependency-name: "@angular/platform-browser"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump css-loader from 5.2.4 to 5.2.6

Bumps [css-loader](https://github.com/webpack-contrib/css-loader) from 5.2.4 to 5.2.6.
- [Release notes](https://github.com/webpack-contrib/css-loader/releases)
- [Changelog](https://github.com/webpack-contrib/css-loader/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack-contrib/css-loader/compare/v5.2.4...v5.2.6)

---
updated-dependencies:
- dependency-name: css-loader
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1603 from matteobruni/dependabot/npm_and_yarn/dev/riotjs/cli-5.3.0

build(deps-dev): bump @riotjs/cli from 5.2.1 to 5.3.0

matteobruni committed:
Merge pull request #1604 from matteobruni/dependabot/npm_and_yarn/dev/angular/platform-browser-12.0.3

build(deps): bump @angular/platform-browser from 12.0.0 to 12.0.3

matteobruni committed:
Merge pull request #1605 from matteobruni/dependabot/npm_and_yarn/dev/css-loader-5.2.6

build(deps-dev): bump css-loader from 5.2.4 to 5.2.6

matteobruni committed:
build: added all presets README.md

dependabot[bot] committed:
build(deps-dev): bump babel-plugin-inferno from 6.2.0 to 6.3.0

Bumps [babel-plugin-inferno](https://github.com/trueadm/babel-plugin-inferno) from 6.2.0 to 6.3.0.
- [Release notes](https://github.com/trueadm/babel-plugin-inferno/releases)
- [Commits](https://github.com/trueadm/babel-plugin-inferno/compare/v6.2.0...6.3.0)

---
updated-dependencies:
- dependency-name: babel-plugin-inferno
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump uglify-js from 3.13.6 to 3.13.9

Bumps [uglify-js](https://github.com/mishoo/UglifyJS) from 3.13.6 to 3.13.9.
- [Release notes](https://github.com/mishoo/UglifyJS/releases)
- [Commits](https://github.com/mishoo/UglifyJS/compare/v3.13.6...v3.13.9)

---
updated-dependencies:
- dependency-name: uglify-js
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1606 from matteobruni/dependabot/npm_and_yarn/dev/babel-plugin-inferno-6.3.0

build(deps-dev): bump babel-plugin-inferno from 6.2.0 to 6.3.0

matteobruni committed:
Merge pull request #1607 from matteobruni/dependabot/npm_and_yarn/dev/uglify-js-3.13.9

build(deps-dev): bump uglify-js from 3.13.6 to 3.13.9

dependabot[bot] committed:
build(deps-dev): bump @babel/plugin-proposal-class-properties

Bumps [@babel/plugin-proposal-class-properties](https://github.com/babel/babel/tree/HEAD/packages/babel-plugin-proposal-class-properties) from 7.13.0 to 7.14.5.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.14.5/packages/babel-plugin-proposal-class-properties)

---
updated-dependencies:
- dependency-name: "@babel/plugin-proposal-class-properties"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1608 from matteobruni/dependabot/npm_and_yarn/dev/babel/plugin-proposal-class-properties-7.14.5

build(deps-dev): bump @babel/plugin-proposal-class-properties from 7.13.0 to 7.14.5

matteobruni committed:
fix: changed click handler, now moved to Container class but kept compatibility

matteobruni committed:
build: updated deps before release

matteobruni committed:
chore(release): published new version

 - ng-particles@2.11.0
 - inferno-particles@1.11.0
 - jquery-particles@1.28.0
 - preact-particles@1.28.0
 - react-tsparticles@1.28.0
 - riot-particles@1.2.0
 - solid-particles@1.11.0
 - svelte-particles@1.11.0
 - particles.vue@2.11.0
 - particles.vue3@1.11.0
 - web-particles@1.1.0
 - tsparticles-editor@1.28.0
 - tsparticles@1.28.0
 - ng-particles-demo@1.13.0
 - editor-demo@1.13.0
 - inferno-particles-demo@1.11.0
 - jquery-particles-demo@1.13.0
 - preact-particles-demo@1.13.0
 - react-tsparticles-demo@1.13.0
 - riot-particles-demo@1.2.0
 - solid-particles-demo@1.11.0
 - svelte-demo@1.11.0
 - tsparticles-demo@1.13.0
 - particles.vue-demo@1.13.0
 - particles.vue3-demo@1.11.0
 - web-particles-demo@1.13.0
 - tsparticles-preset-big-circles@1.13.0
 - tsparticles-preset-bubbles@1.13.0
 - tsparticles-preset-confetti@1.13.0
 - tsparticles-preset-fire@1.13.0
 - tsparticles-preset-firefly@1.13.0
 - tsparticles-preset-fountain@1.13.0
 - tsparticles-preset-links@1.13.0
 - tsparticles-preset-sea-anemone@1.13.0
 - tsparticles-preset-snow@1.13.0
 - tsparticles-preset-stars@1.13.0
 - tsparticles-shape-bubble@1.13.0
 - tsparticles-shape-confetti@1.13.0
 - tsparticles-shape-heart@1.13.0
 - tsparticles-shape-multiline-text@1.13.0
 - tsparticles-shape-rounded-rect@1.13.0
 - tsparticles-shape-spiral@1.13.0
 - cra-template-particles-typescript@1.11.0
 - cra-template-particles@1.11.0
 - tsparticles-website@1.28.0

matteobruni committed:
build: removed tests from riot component

dependabot[bot] committed:
build(deps): bump @angular/core from 12.0.3 to 12.0.4

Bumps [@angular/core](https://github.com/angular/angular/tree/HEAD/packages/core) from 12.0.3 to 12.0.4.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.4/packages/core)

---
updated-dependencies:
- dependency-name: "@angular/core"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
build: updated website

matteobruni committed:
Merge pull request #1595 from matteobruni/v1

1.28.0

matteobruni committed:
Merge branch 'main' into dev

matteobruni committed:
build: update deps

matteobruni committed:
Merge pull request #1612 from matteobruni/dependabot/npm_and_yarn/dev/angular/core-12.0.4

build(deps): bump @angular/core from 12.0.3 to 12.0.4

dependabot[bot] committed:
build(deps): bump riot from 5.4.3 to 5.4.4

Bumps [riot](https://github.com/riot/riot) from 5.4.3 to 5.4.4.
- [Release notes](https://github.com/riot/riot/releases)
- [Commits](https://github.com/riot/riot/compare/v5.4.3...v5.4.4)

---
updated-dependencies:
- dependency-name: riot
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1611 from matteobruni/dependabot/npm_and_yarn/dev/riot-5.4.4

build(deps): bump riot from 5.4.3 to 5.4.4

matteobruni committed:
build: some merge fixes

matteobruni committed:
build: some merge fixes

matteobruni committed:
build: some merge fixes

matteobruni committed:
build: some merge fixes

matteobruni committed:
build: some merge fixes

matteobruni committed:
build: some merge fixes

matteobruni committed:
build: some deps updated

ghost commented:
## DeepCode's analysis on [#e6e1e8](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/5bd8958d2df2879234fc3ea8f2b9548c08a48b76/matteobruni/tsparticles/e6e1e8d18e840738e6deb632d50ea5a403a37677/pr/_/%2F/code/?utm_source=gh_review&c=0&w=3&i=2&) found:

- :warning: **3** warnings, :information_source: **2** minor issues. :point_down:
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">No catch method for promise. This may result in an unhandled promise rejection. Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/e6e1e8d18e840738e6deb632d50ea5a403a37677/components/web/build.ts#L45">build.ts:45</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/5bd8958d2df2879234fc3ea8f2b9548c08a48b76/matteobruni/tsparticles/e6e1e8d18e840738e6deb632d50ea5a403a37677/pr/_/%2Fcomponents%2Fweb%2Fbuild.ts/javascript%2Fdc_interfile_project%2FPromiseNotCaughtNode/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">Consider using csurf middleware for your Express app to protect against CSRF attacks. Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/e6e1e8d18e840738e6deb632d50ea5a403a37677/demo/web/app.js#L5">app.js:5</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/5bd8958d2df2879234fc3ea8f2b9548c08a48b76/matteobruni/tsparticles/e6e1e8d18e840738e6deb632d50ea5a403a37677/pr/_/%2Fdemo%2Fweb%2Fapp.js/javascript%2Fdc_interfile_project%2FUseCsurfForExpress/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">This endpoint handler performs a file system operation and does not use a rate-limiting mechanism. It may enable the attackers to perform Denial-of-service attacks. Consider using a rate-limiting middleware such as express-limit. Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/e6e1e8d18e840738e6deb632d50ea5a403a37677/demo/web/app.js#L20">app.js:20</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/e6e1e8d18e840738e6deb632d50ea5a403a37677/demo/vanilla/app.js#L41">app.js:41</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/e6e1e8d18e840738e6deb632d50ea5a403a37677/demo/vanilla/app.js#L45">app.js:45</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/5bd8958d2df2879234fc3ea8f2b9548c08a48b76/matteobruni/tsparticles/e6e1e8d18e840738e6deb632d50ea5a403a37677/pr/_/%2Fdemo%2Fweb%2Fapp.js/javascript%2Fdc_interfile_project%2FNoRateLimitingForExpensiveWebOperation/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/5bd8958d2df2879234fc3ea8f2b9548c08a48b76/matteobruni/tsparticles/e6e1e8d18e840738e6deb632d50ea5a403a37677/pr/_/%2F/code/?utm_source=gh_review&c=0&w=3&i=2&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=matteobruni)
#### 👉 The DeepCode service and API will be deprecated in August, 2021. [Here](https://www.deepcode.ai/blog/migrate-to-snyk-code) is the information how to migrate. Thank you for using DeepCode 🙏 ❤️ !
If you are using our plugins, you might be interested in their successors: [Snyk's JetBrains plugin](https://snyk.co/udpkq) and [Snyk's VS Code plugin](https://snyk.co/udpkr).

dependabot[bot] committed:
build(deps-dev): bump postcss from 8.3.1 to 8.3.2

Bumps [postcss](https://github.com/postcss/postcss) from 8.3.1 to 8.3.2.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/8.3.1...8.3.2)

---
updated-dependencies:
- dependency-name: postcss
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
build: updated deps

matteobruni committed:
Merge pull request #1616 from matteobruni/dependabot/npm_and_yarn/dev/postcss-8.3.2

build(deps-dev): bump postcss from 8.3.1 to 8.3.2

dependabot[bot] committed:
build(deps-dev): bump cssnano from 4.1.11 to 5.0.2

Bumps [cssnano](https://github.com/cssnano/cssnano) from 4.1.11 to 5.0.2.
- [Release notes](https://github.com/cssnano/cssnano/releases)
- [Commits](https://github.com/cssnano/cssnano/compare/v4.1.11...cssnano@5.0.2)

Signed-off-by: dependabot[bot] <support@github.com>

code-factor committed:
[CodeFactor] Apply fixes

[ci skip] [skip ci]

matteobruni committed:
Merge pull request #1618 from matteobruni/dependabot/npm_and_yarn/dev/cssnano-5.0.2-cf-autofix

Apply fixes from CodeFactor

matteobruni committed:
Merge pull request #1518 from matteobruni/dependabot/npm_and_yarn/dev/cssnano-5.0.2

build(deps-dev): bump cssnano from 4.1.11 to 5.0.2

matteobruni committed:
build: fixed line shape rotation, it was 90 degrees out of phase

matteobruni committed:
Merge remote-tracking branch 'origin/dev' into dev

matteobruni committed:
chore(release): published new version

 - ng-particles@2.11.1
 - inferno-particles@1.11.1
 - jquery-particles@1.28.1
 - preact-particles@1.28.1
 - react-tsparticles@1.28.1
 - riot-particles@1.2.1
 - solid-particles@1.11.1
 - svelte-particles@1.11.1
 - particles.vue@2.11.1
 - particles.vue3@1.11.1
 - web-particles@1.1.1
 - tsparticles-editor@1.28.1
 - tsparticles@1.28.1
 - ng-particles-demo@1.13.1
 - editor-demo@1.13.1
 - inferno-particles-demo@1.11.1
 - jquery-particles-demo@1.13.1
 - preact-particles-demo@1.13.1
 - react-tsparticles-demo@1.13.1
 - riot-particles-demo@1.2.1
 - solid-particles-demo@1.11.1
 - svelte-demo@1.11.1
 - tsparticles-demo@1.13.1
 - particles.vue-demo@1.13.1
 - particles.vue3-demo@1.11.1
 - web-particles-demo@1.13.1
 - tsparticles-preset-big-circles@1.13.1
 - tsparticles-preset-bubbles@1.13.1
 - tsparticles-preset-confetti@1.13.1
 - tsparticles-preset-fire@1.13.1
 - tsparticles-preset-firefly@1.13.1
 - tsparticles-preset-fountain@1.13.1
 - tsparticles-preset-links@1.13.1
 - tsparticles-preset-sea-anemone@1.13.1
 - tsparticles-preset-snow@1.13.1
 - tsparticles-preset-stars@1.13.1
 - tsparticles-shape-bubble@1.13.1
 - tsparticles-shape-confetti@1.13.1
 - tsparticles-shape-heart@1.13.1
 - tsparticles-shape-multiline-text@1.13.1
 - tsparticles-shape-rounded-rect@1.13.1
 - tsparticles-shape-spiral@1.13.1
 - cra-template-particles-typescript@1.11.1
 - cra-template-particles@1.11.1
 - tsparticles-website@1.28.1

dependabot[bot] committed:
build(deps-dev): bump typedoc from 0.21.0-beta.2 to 0.21.0-beta.3

Bumps [typedoc](https://github.com/TypeStrong/TypeDoc) from 0.21.0-beta.2 to 0.21.0-beta.3.
- [Release notes](https://github.com/TypeStrong/TypeDoc/releases)
- [Commits](https://github.com/TypeStrong/TypeDoc/commits)

---
updated-dependencies:
- dependency-name: typedoc
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump webpack from 5.38.1 to 5.39.0

Bumps [webpack](https://github.com/webpack/webpack) from 5.38.1 to 5.39.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.38.1...v5.39.0)

---
updated-dependencies:
- dependency-name: webpack
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
build: prettier

dependabot[bot] committed:
build(deps): bump tslib from 2.2.0 to 2.3.0

Bumps [tslib](https://github.com/Microsoft/tslib) from 2.2.0 to 2.3.0.
- [Release notes](https://github.com/Microsoft/tslib/releases)
- [Commits](https://github.com/Microsoft/tslib/compare/2.2.0...2.3.0)

---
updated-dependencies:
- dependency-name: tslib
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1621 from matteobruni/dependabot/npm_and_yarn/dev/typedoc-0.21.0-beta.3

build(deps-dev): bump typedoc from 0.21.0-beta.2 to 0.21.0-beta.3

matteobruni committed:
Merge pull request #1622 from matteobruni/dependabot/npm_and_yarn/dev/webpack-5.39.0

build(deps-dev): bump webpack from 5.38.1 to 5.39.0

matteobruni committed:
Merge pull request #1624 from matteobruni/dependabot/npm_and_yarn/dev/tslib-2.3.0

build(deps): bump tslib from 2.2.0 to 2.3.0

matteobruni committed:
build: website update

matteobruni committed:
Merge remote-tracking branch 'origin/dev' into dev

matteobruni committed:
build: cleaned some configs

matteobruni committed:
Merge pull request #1620 from matteobruni/dev

1.28.1

dependabot[bot] committed:
build(deps): bump postcss from 7.0.35 to 7.0.36 in /components/angular

Bumps [postcss](https://github.com/postcss/postcss) from 7.0.35 to 7.0.36.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/7.0.35...7.0.36)

---
updated-dependencies:
- dependency-name: postcss
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump postcss from 7.0.35 to 7.0.36 in /components/vue3

Bumps [postcss](https://github.com/postcss/postcss) from 7.0.35 to 7.0.36.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/7.0.35...7.0.36)

---
updated-dependencies:
- dependency-name: postcss
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge branch 'main' into v2_dev

matteobruni committed:
Merge pull request #1625 from matteobruni/dependabot/npm_and_yarn/components/angular/postcss-7.0.36

build(deps): bump postcss from 7.0.35 to 7.0.36 in /components/angular

matteobruni committed:
Merge branch 'main' into dependabot/npm_and_yarn/components/vue3/postcss-7.0.36

matteobruni committed:
Merge pull request #1626 from matteobruni/dependabot/npm_and_yarn/components/vue3/postcss-7.0.36

build(deps): bump postcss from 7.0.35 to 7.0.36 in /components/vue3

dependabot[bot] committed:
build(deps): bump glob-parent from 5.1.1 to 5.1.2 in /components/jquery

Bumps [glob-parent](https://github.com/gulpjs/glob-parent) from 5.1.1 to 5.1.2.
- [Release notes](https://github.com/gulpjs/glob-parent/releases)
- [Changelog](https://github.com/gulpjs/glob-parent/blob/main/CHANGELOG.md)
- [Commits](https://github.com/gulpjs/glob-parent/compare/v5.1.1...v5.1.2)

---
updated-dependencies:
- dependency-name: glob-parent
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1627 from matteobruni/dependabot/npm_and_yarn/components/jquery/glob-parent-5.1.2

build(deps): bump glob-parent from 5.1.1 to 5.1.2 in /components/jquery

dependabot[bot] committed:
build(deps): bump glob-parent from 5.1.1 to 5.1.2 in /components/svelte

Bumps [glob-parent](https://github.com/gulpjs/glob-parent) from 5.1.1 to 5.1.2.
- [Release notes](https://github.com/gulpjs/glob-parent/releases)
- [Changelog](https://github.com/gulpjs/glob-parent/blob/main/CHANGELOG.md)
- [Commits](https://github.com/gulpjs/glob-parent/compare/v5.1.1...v5.1.2)

---
updated-dependencies:
- dependency-name: glob-parent
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump glob-parent from 5.1.1 to 5.1.2 in /components/react

Bumps [glob-parent](https://github.com/gulpjs/glob-parent) from 5.1.1 to 5.1.2.
- [Release notes](https://github.com/gulpjs/glob-parent/releases)
- [Changelog](https://github.com/gulpjs/glob-parent/blob/main/CHANGELOG.md)
- [Commits](https://github.com/gulpjs/glob-parent/compare/v5.1.1...v5.1.2)

---
updated-dependencies:
- dependency-name: glob-parent
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1628 from matteobruni/dependabot/npm_and_yarn/components/svelte/glob-parent-5.1.2

build(deps): bump glob-parent from 5.1.1 to 5.1.2 in /components/svelte

matteobruni committed:
Merge branch 'main' into dependabot/npm_and_yarn/components/react/glob-parent-5.1.2

dependabot[bot] committed:
build(deps): bump glob-parent from 5.1.1 to 5.1.2 in /components/preact

Bumps [glob-parent](https://github.com/gulpjs/glob-parent) from 5.1.1 to 5.1.2.
- [Release notes](https://github.com/gulpjs/glob-parent/releases)
- [Changelog](https://github.com/gulpjs/glob-parent/blob/main/CHANGELOG.md)
- [Commits](https://github.com/gulpjs/glob-parent/compare/v5.1.1...v5.1.2)

---
updated-dependencies:
- dependency-name: glob-parent
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1630 from matteobruni/dependabot/npm_and_yarn/components/preact/glob-parent-5.1.2

build(deps): bump glob-parent from 5.1.1 to 5.1.2 in /components/preact

matteobruni committed:
Merge branch 'main' into dependabot/npm_and_yarn/components/react/glob-parent-5.1.2

matteobruni committed:
Merge pull request #1629 from matteobruni/dependabot/npm_and_yarn/components/react/glob-parent-5.1.2

build(deps): bump glob-parent from 5.1.1 to 5.1.2 in /components/react

matteobruni committed:
Merge branch 'main' into v2_dev

matteoums committed:
aligning some files from v1

matteobruni committed:
build: fix out of canvas updater

matteobruni committed:
Merge branch 'v2_dev' of https://github.com/matteobruni/tsparticles into v2_dev
